libtransact/src/protocol/errors.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2019 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[derive(Debug, PartialEq)]
+pub enum BuilderError {
+    MissingField(String),
+}
+
+impl std::fmt::Display for BuilderError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match *self {
+            BuilderError::MissingField(ref s) => write!(f, "MissingField: {}", s),
+        }
+    }
+}
+
+pub trait Build {
```

Comment:
Looks like you have the trait in this file that is also included in another commit/separate file.